WireDomain/Sources/WireDomain/Synchronization/PullPendingUpdateEventsSync.swift

```diff
@@ -78,30 +79,46 @@ public struct PullPendingUpdateEventsSync: PullPendingUpdateEventsSyncProtocol {
                 )
             }
 
-            for envelope in envelopes {
-                count += 1
+            // We'll insert new events from this index.
+            var currentIndex = try await store.indexOfLastEventEnvelope() + 1
 
-                log("decrypting...", envelopeID: envelope.id)
-                var decryptedEnvelope = envelope
-                let decryptedEvents = try await decryptor.decryptEvents(in: envelope)
-                decryptedEnvelope.events = decryptedEvents
+            // We are decrypting the batch within one core crypto transaction
+            try await coreCryptoProvider.coreCrypto().perform { context in
 
-                log("storing...", envelopeID: envelope.id)
-                try await store.persistEventEnvelope(
-                    decryptedEnvelope,
-                    index: currentIndex
-                )
+                var lastEnvelopeID: UUID?
+                var decryptedEnvelopes: [UpdateEventEnvelope] = []
 
-                events.append(contentsOf: decryptedEvents)
+                for envelope in envelopes {
+                    count += 1
```

Comment:
I don't see the `count` variable being read after this line. Should it be used to increment `currentIndex`?

Reply:
It's read in the `log()` function scoped to this function. `currentIndex` doesn't need to updated since we are fetching it from the repository.